src/main/java/soot/jimple/toolkits/scalar/UnreachableCodeEliminator.java

```diff
@@ -99,9 +101,16 @@ protected void internalTransform(Body body, String phaseName, Map<String, String
     // occur in practice, and certainly no in code generated from Java source.
     final Chain<Trap> traps = body.getTraps();
     for (Iterator<Trap> it = traps.iterator(); it.hasNext();) {
-      Trap trap = it.next();
-      if ((trap.getBeginUnit() == trap.getEndUnit()) || !reachable.contains(trap.getHandlerUnit())) {
+      final Trap trap = it.next();
+      UnitBox beginBox = trap.getBeginUnitBox();
+      UnitBox endBox = trap.getEndUnitBox();
+      UnitBox handlerBox = trap.getHandlerUnitBox();
+      if ((beginBox.getUnit() == endBox.getUnit()) || !reachable.contains(handlerBox.getUnit())) {
```

Comment:
please restructure the UnitBox assignments inside the if block so the assignments are just executed when needed

Reply:
I don't think any improvement can be made. The first two (beginBox and endBox) will always be used since they are in the first condition. The final variable (handlerBox) will always be used as well. If the first condition is true, it will be used inside the if statement body and if the first condition is false, it will be used in the second condition (and possibly again inside the body).